# Processing X-matched Data

## Loading Data

In [1]:
import pandas as pd
from astropy import units as u
from tqdm import tqdm

In [7]:
full_df

RA        DEC  MAG_APER_4_G  MAG_APER_4_R  \
COADD_OBJECT_ID                                                      
1503227012        71.322239 -64.162782     16.440767     17.171459   
1605417922        93.869829 -38.353100     16.069118     16.261015   
951955941        333.051323 -53.690433     15.718325     16.002075   
1294498362        38.768114 -11.054056     16.528463     15.968455   
1146658119        18.076905  -4.611584     16.093866     16.410271   
...                     ...        ...           ...           ...   
1247372205        32.345135 -58.579808     25.462448     23.163431   
883878897        312.909545 -46.126603     26.141560     24.154366   
926762526        323.800579 -52.293027     26.859566     24.696547   
1455214085        61.453315 -28.530423     25.501448     24.546787   
1444823577        49.983829 -39.893271     99.000000     25.309364   

                 MAG_APER_4_I  MAG_APER_4_Z  MAG_APER_4_Y  MAGERR_APER_4_G  \
COADD_OBJECT_ID                                                              
1503227012          16.432383     15.318611     15.230842         0.000414   
1605417922          16.570312     15.371555     15.144802         0.000395   
951955941           16.629000     15.392603     14.970550         0.000234   
1294498362          16.413517     15.402062     15.346148         0.000348   
1146658119          16.581682     15.408277     15.167622         0.000239   
...                       ...           ...           ...              ...   
1247372205          22.000992     20.999535     20.868917         0.235969   
883878897           22.023184     20.999550     20.734533         0.372111   
926762526           22.041519     20.999550     20.822565         0.870667   
1455214085          22.136644     20.999723     20.509132         0.183691   
1444823577          22.615095     20.999758     20.518101        99.000000   

                 MAGERR_APER_4_R  MAGERR_APER_4_I  ...  Hperrbits   Kspmag  \
COADD_OBJECT_ID                                    ...                       
1503227012              0.000558         0.000421  ...          0  13.4174   
1605417922              0.000399         0.000524  ...          0  13.5606   
951955941               0.000285         0.000398  ...       4096  13.1156   
1294498362              0.000236         0.000322  ...          0  13.3234   
1146658119              0.000337         0.000451  ...          0  12.7759   
...                          ...              ...  ...        ...      ...   
1247372205              0.034844         0.020897  ...          0  16.6428   
883878897               0.088149         0.022554  ...    8388736  18.4054   
926762526               0.170713         0.022444  ...          0  17.8292   
1455214085              0.096168         0.018437  ...          0  17.7694   
1444823577              0.264413         0.043572  ...          0  17.1611   

                e_Kspmag    Ksap3  e_Ksap3   Ksapc3  Ksavgap3  Ksperrbits  \
COADD_OBJECT_ID                                                             
1503227012        0.0066  13.4094   0.0035  13.7183   106.749           0   
1605417922        0.0087  13.5753   0.0045  13.8835    47.765     8388736   
951955941         0.0047  13.2704   0.0032  13.6313   104.815        4112   
1294498362        0.0039  13.2586   0.0027  13.4323   102.671           0   
1146658119        0.0029  12.7853   0.0019  13.0428   104.308           0   
...                  ...      ...      ...      ...       ...         ...   
1247372205        0.1597  17.3496   0.0747  17.5481    96.917           0   
883878897         0.4778  18.1871   0.1819  18.5382    70.839     8388736   
926762526         0.2729  18.0389   0.1873  18.4879   104.348        4096   
1455214085        0.2953  17.8690   0.1343  18.2419   101.190           0   
1444823577        0.1941  17.3036   0.1006  17.8167   102.008           0   

                 prim_x  angDist_x  
COADD_OBJECT_ID                     
15032270

In [15]:
full_df = pd.read_csv('../data/interim/des_wise_vhs_objects.csv', index_col=0)

In [18]:
old_names = [
    'RA','DEC',
    'MAG_APER_4_G','MAGERR_APER_4_G',
    'MAG_APER_4_R','MAGERR_APER_4_R',
    'MAG_APER_4_I','MAGERR_APER_4_I',
    'MAG_APER_4_Z','MAGERR_APER_4_Z',
    'MAG_APER_4_Y','MAGERR_APER_4_Y',
    'Jpmag','e_Jpmag',
    'Kspmag','e_Kspmag',
    'FW1lbs','e_FW1lbs',
    'FW2lbs','e_FW2lbs'
]
new_names = [
    'ra_des','dec_des',
    'g_mag','g_magerr',
    'r_mag','r_magerr',
    'i_mag','i_magerr',
    'z_mag','z_magerr',
    'Y_mag','Y_magerr',
    'J_mag','J_magerr',
    'K_mag','K_magerr',
    'W1_flux','W1_fluxerr',
    'W2_flux','W2_fluxerr'
]
neat_df = full_df[old_names].copy()
neat_df.rename(columns = dict(zip(old_names, new_names)), inplace=True)
neat_df

ra_des    dec_des      g_mag   g_magerr      r_mag  \
COADD_OBJECT_ID                                                           
1503227012        71.322239 -64.162782  16.440767   0.000414  17.171459   
1605417922        93.869829 -38.353100  16.069118   0.000395  16.261015   
951955941        333.051323 -53.690433  15.718325   0.000234  16.002075   
1294498362        38.768114 -11.054056  16.528463   0.000348  15.968455   
1146658119        18.076905  -4.611584  16.093866   0.000239  16.410271   
...                     ...        ...        ...        ...        ...   
1247372205        32.345135 -58.579808  25.462448   0.235969  23.163431   
883878897        312.909545 -46.126603  26.141560   0.372111  24.154366   
926762526        323.800579 -52.293027  26.859566   0.870667  24.696547   
1455214085        61.453315 -28.530423  25.501448   0.183691  24.546787   
1444823577        49.983829 -39.893271  99.000000  99.000000  25.309364   

                 r_magerr      i_mag  i_magerr      z_mag  z_magerr  \
COADD_OBJECT_ID                                                       
1503227012       0.000558  16.432383  0.000421  15.318611  0.000304   
1605417922       0.000399  16.570312  0.000524  15.371555  0.000274   
951955941        0.000285  16.629000  0.000398  15.392603  0.000276   
1294498362       0.000236  16.413517  0.000322  15.402062  0.000244   
1146658119       0.000337  16.581682  0.000451  15.408277  0.000376   
...                   ...        ...       ...        ...       ...   
1247372205       0.034844  22.000992  0.020897  20.999535  0.022301   
883878897        0.088149  22.023184  0.022554  20.999550  0.016858   
926762526        0.170713  22.041519  0.022444  20.999550  0.015904   
1455214085       0.096168  22.136644  0.018437  20.999723  0.012471   
1444823577       0.264413  22.615095  0.043572  20.999758  0.021789   

                     Y_mag  Y_magerr    J_mag  J_magerr    K_mag  K_magerr  \
COADD_OBJECT_ID                                                              
1503227012       15.230842  0.000759  13.9112    0.0020  13.4174    0.0066   
1605417922       15.144802  0.000660  13.9764    0.0022  13.5606    0.0087   
951955941        14.970550  0.000444  13.6782    0.0022  13.1156    0.0047   
1294498362       15.346148  0.000598  13.8421    0.0018  13.3234    0.0039   
1146658119       15.167622  0.000423  13.3455    0.0015  12.7759    0.0029   
...                    ...       ...      ...       ...      ...       ...   
1247372205       20.868917  0.053797  18.2481    0.1171  16.6428    0.1597   
883878897        20.734533  0.045661  18.6226    0.1918  18.4054    0.4778   
926762526        20.822565  0.045743  18.8938    0.1509  17.8292    0.2729   
1455214085       20.509132  0.025710  18.6371    0.1025  17.7694    0.2953   
1444823577       20.518101  0.035152  18.3440    0.0886  17.1611    0.1941   

                   W1_flux  W1_fluxerr   W2_flux  W2_fluxerr  
COADD_OBJECT_ID                                               
1503227012       4502.1100    11.97190  4153.060     27.9172  
1605417922       3952.0600    12.32280  3792.910     30.6441  
951955941        5120.6100    14.13130  4870.040     37.2086  
1294498362       5113.8200    14.41800  4693.550     37.7026  
1146658119       8410.9000    17.48560  7985.420     44.9709  
...                    ...         ...       ...         ...  
1247372205        496.0320     6.33226   652.842     22.0617  
883878897          68.9175     6.96753     0.000      0.0000  
926762526          76.4045     7.01943     0.000      0.0000  
1455214085        106.6550     6.23375     0.000      0.0000  
1444823577        214.8860     6.05167   249.775     22.2626  

[151629 rows x 20 columns]

In [31]:
# Renaming columns for convenience
neat_df = main_df.rename(columns={
    "t1_t1_ra":"ra_des",
    "t1_t1_dec":"dec_des",
    "t1_t1_mag_aper_4_g":"g_mag",
    "t1_t1_mag_aper_4_r":"r_mag",
    "t1_t1_mag_aper_4_i":"i_mag",
    "t1_t1_mag_aper_4_z":"z_mag",
    "t1_t1_mag_aper_4_y":"Y_mag",
    "t1_t1_magerr_aper_4_g":"g_magerr",
    "t1_t1_magerr_aper_4_r":"r_magerr",
    "t1_t1_magerr_aper_4_i":"i_magerr",
    "t1_t1_magerr_aper_4_z":"z_magerr",
    "t1_t1_magerr_aper_4_y":"Y_magerr",
    "japermag4":"J_mag_vg",
    "japermag4err":"J_magerr",
    "ksapermag4":"K_mag_vg",
    "ksapermag4err":"K_magerr",
    "t1_mag_w1_vg":"W1_mag_vg",
    "t1_mag_w2_vg":"W2_mag_vg",
    "t1_fluxlbs_w1":"W1_flux_nMgy",
    "t1_fluxlbs_w2":"W2_flux_nMgy",
    "t1_dfluxlbs_w1":"W1_fluxerr_nMgy",
    "t1_dfluxlbs_w2":"W2_fluxerr_nMgy",
    "ra2000":"ra_vhs",
    "dec2000":"dec_vhs",
    "dist_arcsec":"dist_des_vhs",
    "t1_ra":"ra_wise",
    "t1_dec":"dec_wise",
    "t1_dist_arcsec":"dist_des_wise"
})
neat_df.index.names=['coadd_object_id']
neat_df

ra_des    dec_des      g_mag      r_mag      i_mag  \
coadd_object_id                                                           
914697722        319.119200 -45.090354  18.125549  16.507605  17.101320   
1246635859        32.710920 -12.110579  17.643240  16.183584  16.517576   
984527913        344.403976 -40.240537  15.945573  16.843105  16.362082   
921966171        321.279299 -55.389702  17.825924  16.296925  16.691380   
870486068        299.876297 -53.999905  15.880921  16.138315  16.533918   
...                     ...        ...        ...        ...        ...   
890672347        307.551962 -49.679314  28.945040  24.445572  22.015619   
1117460595        12.880701 -45.886776  26.059227  24.326349  22.029612   
1311137024        25.655443  -0.712265  24.325096  23.013109  22.148762   
1437291121        60.389445 -14.630045  26.888927  24.858255  22.059185   
1580230294        86.758839 -46.337940  24.081594  22.841181  22.092491   

                     z_mag      Y_mag  g_magerr  r_magerr  i_magerr  z_magerr  \
coadd_object_id                                                                 
914697722        15.075032  14.713349  0.000741  0.000339  0.000560  0.000400   
1246635859       15.214206  14.503883  0.000647  0.000289  0.000469  0.000453   
984527913        15.239571  15.196484  0.000301  0.000591  0.000548  0.000297   
921966171        15.280548  14.936352  0.000651  0.000296  0.000500  0.000335   
870486068        15.384865  15.597589  0.000244  0.000255  0.000429  0.000242   
...                    ...        ...       ...       ...       ...       ...   
890672347        20.999533  20.731770  5.077159  0.105080  0.019959  0.017075   
1117460595       20.999763  20.902056  0.355723  0.092730  0.020485  0.016868   
1311137024       20.999807  21.066023  0.072679  0.027314  0.023289  0.014506   
1437291121       20.999891  20.687372  0.625257  0.124696  0.016738  0.012384   
1580230294       20.999891  21.399715  0.052119  0.022123  0.020236  0.014069   

                 Y_magerr   dec_wise  W1_fluxerr_nMgy  W2_fluxerr_nMgy  \
coadd_object_id                                                          
914697722        0.000381 -45.090322        26.233025        62.808285   
1246635859       0.000547 -12.110539        24.390047        59.220913   
984527913        0.000755 -40.240554        13.105026        35.000120   
921966171        0.000474 -55.389683        19.049904        46.321644   
870486068        0.001358 -53.999912        12.821824        34.966680   
...                   ...        ...              ...              ...   
890672347        0.047391 -49.679354         7.069963         0.000000   
1117460595       0.040521 -45.886752         6.069972         0.000000   
1311137024       0.052359  -0.712180         7.968876        32.033960   
1437291121       0.035437 -14.629854         7.172823         0.000000   
1580230294       0.063653 -46.337891         5.628012        20.959763   

                 W1_flux_nMgy  W2_flux_nMgy  W1_mag_vg  W2_mag_vg     ra_wise  \
coadd_object_id                                                                 
914697722         23075.97900   26846.29100  11.592440  11.428191  319.119198   
1246635859        19631.18200   23035.37100  11.768310  11.594521   32.710872   
984527913          4355.65600    4089.07280  13.403144  13.471869  344.403963   
921966171         11148.60700   11932.68000  12.382002  12.308417  321.279281   
870486068          3824.09700    3655.99150  13.542056  13.595168  299.876326   
...                       ...           ...        ...        ...         ...   
890672347            67.90782       0.00000  17.809786        inf  307.551566   
1117460595           66.67445       0.00000  17.847363        inf   12.880809   
1311137024          276.67416     365.60960  16.391867  16.088333   25.655491   
1437291121           51.82774       0.00000  18.162922        inf   60.389366   
1580230294          331.80670     865.44507  16.192630  

### Converting Magnitudes to Flux
Useful for certain SED fitting programs

In [32]:
def mag_to_flx(mag):
    """
    Input: an AB magnitude
    Output: a flux in microjansky (muJy)
    Works with np arrays and pd Series
    """
    return 10**(29-(48.60/2.5)) * 10**(-mag/2.5)

print(mag_to_flx(0) * 1e-6) # 3631 Jy

3630.7805477010024


### Errors in Magnitude and Flux
The errors use the following relation:
$$
m = -2.5\log(f/f_0)
\qquad\Rightarrow\qquad
|\text{d}m| = 2.5 \log(e) \frac{\text{d}f}{f} \approx 1.09 \frac{\text{d}f}{f}
$$
For DES and VHS, $\text{d}m$ is given; for WISE, $\text{d}f$ and $f$ are given (both in nanomaggies, which seem to be simply proportional to Jy, see [here](https://www.sdss3.org/dr8/algorithms/magnitudes.php)).

### Converting Vega to AB

The VHS and WISE magnitudes are in the Vega magnitude system; the AB system corresponds better to $F_\nu$. $m_{AB} = m_{Vega} + \Delta m$ where $\Delta m$ depends on the band (see specs for [VHS](https://www.eso.org/rm/api/v1/public/releaseDescriptions/144) and [WISE](https://wise2.ipac.caltech.edu/docs/release/allsky/expsup/sec4_4h.html)):

|Band|$\Delta m$|
|---|---|
|J|0.916|
|K|1.827|
|W1|2.699|
|W2|3.339|

In [33]:
for band in ['W1','W2']:
    neat_df[f'{band}_magerr'] = 2.5*np.log10(np.e) * neat_df[f'{band}_fluxerr_nMgy'] / neat_df[f'{band}_flux_nMgy']

delta_m = {'J':0.916,
           'K':1.827,
           'W1':2.699,
           'W2':3.339}
for band in ['J','K','W1','W2']:
    neat_df[f'{band}_mag'] = neat_df[f'{band}_mag_vg'] + delta_m[band]        

all_bands = ['g','r','i','z','Y','J','K', 'W1','W2']
for band in all_bands:
    flx = mag_to_flx(neat_df[f'{band}_mag'])
    neat_df[f'{band}_flux'] = flx
    neat_df[f'{band}_fluxerr'] = flx*neat_df[f'{band}_magerr'] / (2.5*np.log10(np.e))

neat_df

ra_des    dec_des      g_mag      r_mag      i_mag  \
coadd_object_id                                                           
914697722        319.119200 -45.090354  18.125549  16.507605  17.101320   
1246635859        32.710920 -12.110579  17.643240  16.183584  16.517576   
984527913        344.403976 -40.240537  15.945573  16.843105  16.362082   
921966171        321.279299 -55.389702  17.825924  16.296925  16.691380   
870486068        299.876297 -53.999905  15.880921  16.138315  16.533918   
...                     ...        ...        ...        ...        ...   
890672347        307.551962 -49.679314  28.945040  24.445572  22.015619   
1117460595        12.880701 -45.886776  26.059227  24.326349  22.029612   
1311137024        25.655443  -0.712265  24.325096  23.013109  22.148762   
1437291121        60.389445 -14.630045  26.888927  24.858255  22.059185   
1580230294        86.758839 -46.337940  24.081594  22.841181  22.092491   

                     z_mag      Y_mag  g_magerr  r_magerr  i_magerr  z_magerr  \
coadd_object_id                                                                 
914697722        15.075032  14.713349  0.000741  0.000339  0.000560  0.000400   
1246635859       15.214206  14.503883  0.000647  0.000289  0.000469  0.000453   
984527913        15.239571  15.196484  0.000301  0.000591  0.000548  0.000297   
921966171        15.280548  14.936352  0.000651  0.000296  0.000500  0.000335   
870486068        15.384865  15.597589  0.000244  0.000255  0.000429  0.000242   
...                    ...        ...       ...       ...       ...       ...   
890672347        20.999533  20.731770  5.077159  0.105080  0.019959  0.017075   
1117460595       20.999763  20.902056  0.355723  0.092730  0.020485  0.016868   
1311137024       20.999807  21.066023  0.072679  0.027314  0.023289  0.014506   
1437291121       20.999891  20.687372  0.625257  0.124696  0.016738  0.012384   
1580230294       20.999891  21.399715  0.052119  0.022123  0.020236  0.014069   

                 Y_magerr   dec_wise  W1_fluxerr_nMgy  W2_fluxerr_nMgy  \
coadd_object_id                                                          
914697722        0.000381 -45.090322        26.233025        62.808285   
1246635859       0.000547 -12.110539        24.390047        59.220913   
984527913        0.000755 -40.240554        13.105026        35.000120   
921966171        0.000474 -55.389683        19.049904        46.321644   
870486068        0.001358 -53.999912        12.821824        34.966680   
...                   ...        ...              ...              ...   
890672347        0.047391 -49.679354         7.069963         0.000000   
1117460595       0.040521 -45.886752         6.069972         0.000000   
1311137024       0.052359  -0.712180         7.968876        32.033960   
1437291121       0.035437 -14.629854         7.172823         0.000000   
1580230294       0.063653 -46.337891         5.628012        20.959763   

                 W1_flux_nMgy  W2_flux_nMgy  W1_mag_vg  W2_mag_vg     ra_wise  \
coadd_object_id                                                                 
914697722         23075.97900   26846.29100  11.592440  11.428191  319.119198   
1246635859        19631.18200   23035.37100  11.768310  11.594521   32.710872   
984527913          4355.65600    4089.07280  13.403144  13.471869  344.403963   
921966171         11148.60700   11932.68000  12.382002  12.308417  321.279281   
870486068          3824.09700    3655.99150  13.542056  13.595168  299.876326   
...                       ...           ...        ...        ...         ...   
890672347            67.90782       0.00000  17.809786        inf  307.551566   
1117460595           66.67445       0.00000  17.847363        inf   12.880809   
1311137024          276.67416     365.60960  16.391867  16.088333   25.655491   
1437291121           51.82774       0.00000  18.162922        inf   60.389366   
1580230294          331.80670     865.44507  16.192630  

### Flooring objects below $3\sigma$ detection
If an object is detected *below* a $3\sigma$ level, the magnitude and error are set to `np.inf`, the flux is set to 0, and the flux error is set to the 3$\sigma$ sensitivity. The sensitivities of the three data sources (DES, VHS, and WISE) are provided in different ways:

#### DES

The 10$\sigma$ AB magnitude depths are given [here](https://iopscience.iop.org/article/10.3847/1538-4365/ac00b3/pdf) in Table 3 Row 3:

|Band|$10\sigma$ Sensitivity / mag$_{AB}$|
|---|---|
|g|24.7|
|r|24.4|
|i|23.8|
|z|23.1|
|Y|21.7|

The relationship between magnitudes at different significance levels is
\begin{align*}
m_{n_1\sigma} - m_{n_2\sigma} 
&= -2.5\log(F_{n_1\sigma} / F_{n_2\sigma})\\
&\equiv -2.5 \log(n_1/n_2)\\
\Rightarrow m_{n_1\sigma}
&= m_{n_2\sigma} - 2.5 \log(n_1/n_2)
\end{align*}

So $m_{3\sigma} = m_{10\sigma} - 2.5 \log(3/10)$.

#### VHS

The (median) $5\sigma$ depths are given [here](http://www.eso.org/rm/api/v1/public/releaseDescriptions/144), just above Table 4, as J:20.8, K:20.0. These are $5\sigma$, so $m_{3\sigma} = m_{5\sigma} - 2.5\log(3/5)$

#### WISE

The $m_{n\sigma}$ are not given for each band, but the $5\sigma$ *flux* sensitivities are:

W1: $F_{\nu, 5\sigma} = 0.08$ mJy

W2: $F_{\nu, 5\sigma} = 0.11$ mJy

In [34]:
des_m_3s = np.array([24.7,24.4,23.8,23.1,21.7]) - 2.5*np.log10(3/10)
des_F_3s = mag_to_flx(des_m_3s)
vhs_m_3s = np.array([20.8,20.0]) - 2.5*np.log10(3/5)
vhs_F_3s = mag_to_flx(vhs_m_3s)
wise_F_3s = np.array([0.08e-3, 0.11e-3])
F_3s = {
    'g':des_F_3s[0],
    'r':des_F_3s[1],
    'i':des_F_3s[2],
    'z':des_F_3s[3],
    'Y':des_F_3s[4],
    'J':vhs_F_3s[0],
    'K':vhs_F_3s[1],
    'W1':wise_F_3s[0],
    'W2':wise_F_3s[1]
}

for idd in tqdm(neat_df.index):
    for band in all_bands:
        sensitivity = F_3s[band]
        if neat_df.at[idd, f'{band}_flux'] < sensitivity:
            neat_df.at[idd, f'{band}_mag'] = 99.
            neat_df.at[idd, f'{band}_magerr'] = 99.
            neat_df.at[idd, f'{band}_flux'] = 0.
            neat_df.at[idd, f'{band}_fluxerr'] = sensitivity
neat_df

100%|████████████████████████████████████████████████████████████████████████| 151629/151629 [00:55<00:00, 2753.68it/s]


ra_des    dec_des      g_mag      r_mag      i_mag  \
coadd_object_id                                                           
914697722        319.119200 -45.090354  18.125549  16.507605  17.101320   
1246635859        32.710920 -12.110579  17.643240  16.183584  16.517576   
984527913        344.403976 -40.240537  15.945573  16.843105  16.362082   
921966171        321.279299 -55.389702  17.825924  16.296925  16.691380   
870486068        299.876297 -53.999905  15.880921  16.138315  16.533918   
...                     ...        ...        ...        ...        ...   
890672347        307.551962 -49.679314  99.000000  24.445572  22.015619   
1117460595        12.880701 -45.886776  99.000000  24.326349  22.029612   
1311137024        25.655443  -0.712265  24.325096  23.013109  22.148762   
1437291121        60.389445 -14.630045  99.000000  24.858255  22.059185   
1580230294        86.758839 -46.337940  24.081594  22.841181  22.092491   

                     z_mag      Y_mag   g_magerr  r_magerr  i_magerr  \
coadd_object_id                                                        
914697722        15.075032  14.713349   0.000741  0.000339  0.000560   
1246635859       15.214206  14.503883   0.000647  0.000289  0.000469   
984527913        15.239571  15.196484   0.000301  0.000591  0.000548   
921966171        15.280548  14.936352   0.000651  0.000296  0.000500   
870486068        15.384865  15.597589   0.000244  0.000255  0.000429   
...                    ...        ...        ...       ...       ...   
890672347        20.999533  20.731770  99.000000  0.105080  0.019959   
1117460595       20.999763  20.902056  99.000000  0.092730  0.020485   
1311137024       20.999807  21.066023   0.072679  0.027314  0.023289   
1437291121       20.999891  20.687372  99.000000  0.124696  0.016738   
1580230294       20.999891  21.399715   0.052119  0.022123  0.020236   

                 z_magerr  Y_magerr   dec_wise  W1_fluxerr_nMgy  \
coadd_object_id                                                   
914697722        0.000400  0.000381 -45.090322        26.233025   
1246635859       0.000453  0.000547 -12.110539        24.390047   
984527913        0.000297  0.000755 -40.240554        13.105026   
921966171        0.000335  0.000474 -55.389683        19.049904   
870486068        0.000242  0.001358 -53.999912        12.821824   
...                   ...       ...        ...              ...   
890672347        0.017075  0.047391 -49.679354         7.069963   
1117460595       0.016868  0.040521 -45.886752         6.069972   
1311137024       0.014506  0.052359  -0.712180         7.968876   
1437291121       0.012384  0.035437 -14.629854         7.172823   
1580230294       0.014069  0.063653 -46.337891         5.628012   

                 W2_fluxerr_nMgy  W1_flux_nMgy  W2_flux_nMgy  W1_mag_vg  \
coadd_object_id                                                           
914697722              62.808285   23075.97900   26846.29100  11.592440   
1246635859             59.220913   19631.18200   23035.37100  11.768310   
984527913              35.000120    4355.65600    4089.07280  13.403144   
921966171              46.321644   11148.60700   11932.68000  12.382002   
870486068              34.966680    3824.09700    3655.99150  13.542056   
...                          ...           ...           ...        ...   
890672347               0.000000      67.90782       0.00000  17.809786   
1117460595              0.000000      66.67445       0.00000  17.847363   
1311137024             32.033960     276.67416     365.60960  16.391867   
1437291121              0.000000      51.82774       0.00000  18.162922   
1580230294             20.959763     331.80670     865.44507  16.192630   

                 W2_mag_vg     ra_wise  dist_des_wise    dec_vhs   J_mag_vg  \
coadd_object_id                                                               
914697722        11.428191  319.119198       0.117144 -45.090308  12.526873   
1246635859       11.594

## Cuts
The above dataframe represents a liberal cut in DES data and an X-match (1") to VHS and WISE detections. We begin with 151,629 items.

We then make the following cuts:

* (W1, W2) $\neq 99$. This ensures that the object was in fact detected in WISE.
* (dW1, dW2) $<2.5\log (e)/3$. This ensures a signal-to-noise ratio of greater than 3.
* W1 - W2 $>0.5$ (Vega mags). This removes many of the contaminants (e.g. dwarf stars).
* (J, K) $\neq$ NaN. I'm not too sure why VISTA would do this, but it's not ideal so let's remove them

7,102 objects remain.

In [35]:
cut_df = neat_df[
    (neat_df['W1_mag']!=99.) &
    (neat_df['W2_mag']!=99.) &
    (neat_df['W1_magerr'] < 2.5*np.log10(np.e) / 3) & 
    (neat_df['W2_magerr'] < 2.5*np.log10(np.e) / 3) &
    (neat_df['W1_mag_vg'] - neat_df['W2_mag_vg'] > 0.5) & 
    (np.isnan(neat_df['J_mag']) == False) & 
    (np.isnan(neat_df['K_mag']) == False)
]
cut_df

ra_des    dec_des      g_mag      r_mag      i_mag  \
coadd_object_id                                                           
931684428        324.850726 -42.589015  16.599474  16.393660  16.711504   
1556041733        81.563282 -41.218403  16.374800  16.465633  17.463329   
979002481        342.171547 -51.164810  15.673114  16.068356  17.093698   
995403105        347.111985 -53.411194  17.393402  16.070789  17.274448   
979097465        342.171892 -45.439281  16.905516  16.111046  17.142202   
...                     ...        ...        ...        ...        ...   
887791189        304.571526 -64.443206  99.000000  25.289423  22.204578   
1697986932        79.667443 -54.549900  99.000000  24.895716  22.037481   
1011682001       352.995897 -51.519823  24.987549  23.203817  22.057419   
959369723        335.415977 -47.871207  23.950171  22.853050  22.039396   
1580230294        86.758839 -46.337940  24.081594  22.841181  22.092491   

                     z_mag      Y_mag   g_magerr  r_magerr  i_magerr  \
coadd_object_id                                                        
931684428        15.676861  15.332111   0.000311  0.000273  0.000417   
1556041733       15.783970  15.415310   0.000273  0.000299  0.000593   
979002481        15.902316  15.743353   0.000322  0.000266  0.000554   
995403105        15.892651  14.998402   0.000491  0.000231  0.000594   
979097465        16.109114  15.093253   0.000371  0.000255  0.000554   
...                    ...        ...        ...       ...       ...   
887791189        20.991449  20.781302  99.000000  0.264862  0.028434   
1697986932       20.994463  20.771387  99.000000  0.151750  0.019640   
1011682001       20.994617  20.924448   0.146228  0.040612  0.024272   
959369723        20.998436  20.973963   0.056980  0.027209  0.023609   
1580230294       20.999891  21.399715   0.052119  0.022123  0.020236   

                 z_magerr  Y_magerr   dec_wise  W1_fluxerr_nMgy  \
coadd_object_id                                                   
931684428        0.000259  0.000570 -42.589000        30.999132   
1556041733       0.000254  0.000541 -41.218417        13.246780   
979002481        0.000335  0.000668 -51.164813        29.458530   
995403105        0.000374  0.000368 -53.411204        21.238377   
979097465        0.000358  0.000575 -45.439294        19.046530   
...                   ...       ...        ...              ...   
887791189        0.017894  0.054016 -64.443417         6.983442   
1697986932       0.014022  0.038908 -54.549814         4.191251   
1011682001       0.018400  0.047916 -51.519773         7.501736   
959369723        0.014383  0.047931 -47.871210         7.654265   
1580230294       0.014069  0.063653 -46.337891         5.628012   

                 W2_fluxerr_nMgy  W1_flux_nMgy  W2_flux_nMgy  W1_mag_vg  \
coadd_object_id                                                           
931684428              95.513985  33422.754000   74348.87500  11.189637   
1556041733             33.451920   3525.532200    5923.62200  13.637177   
979002481              95.613100  30659.105000   77761.86000  11.282861   
995403105              52.965280  10385.307000   18389.44100  12.462234   
979097465              46.550224   6881.079600   12015.87800  12.910409   
...                          ...           ...           ...        ...   
887791189              26.935120     76.591940     153.11038  17.715685   
1697986932             16.157791     53.300095      95.89126  18.161716   
1011682001             26.374840    393.562230     723.80360  16.007532   
959369723              28.458288    314.657070     656.23520  16.243427   
1580230294             20.959763    331.806700     865.44507  16.192630   

                 W2_mag_vg     ra_wise  dist_des_wise    dec_vhs   J_mag_vg  \
coadd_object_id                                                               
931684428        10.321410  324.850690       0.109317 -42.588989  13.917383   
1556041733       13.065

In [36]:
# Rearranging the table and removing irrelevant fields
neat_cut_df = cut_df[[
    'ra_des', 'dec_des',
    'g_mag','g_magerr','g_flux','g_fluxerr',
    'r_mag','r_magerr','r_flux','r_fluxerr',
    'i_mag','i_magerr','i_flux','i_fluxerr',
    'z_mag','z_magerr','z_flux','z_fluxerr',
    'Y_mag','Y_magerr','Y_flux','Y_fluxerr',
    'J_mag','J_magerr','J_flux','J_fluxerr',
    'K_mag','K_magerr','K_flux','K_fluxerr',
    'W1_mag','W1_magerr','W1_flux','W1_fluxerr',
    'W2_mag','W2_magerr','W2_flux','W2_fluxerr'
]]
neat_cut_df

ra_des    dec_des      g_mag   g_magerr       g_flux  \
coadd_object_id                                                             
931684428        324.850726 -42.589015  16.599474   0.000311   832.166867   
1556041733        81.563282 -41.218403  16.374800   0.000273  1023.481762   
979002481        342.171547 -51.164810  15.673114   0.000322  1953.234884   
995403105        347.111985 -53.411194  17.393402   0.000491   400.533790   
979097465        342.171892 -45.439281  16.905516   0.000371   627.760134   
...                     ...        ...        ...        ...          ...   
887791189        304.571526 -64.443206  99.000000  99.000000     0.000000   
1697986932        79.667443 -54.549900  99.000000  99.000000     0.000000   
1011682001       352.995897 -51.519823  24.987549   0.146228     0.367266   
959369723        335.415977 -47.871207  23.950171   0.056980     0.954843   
1580230294        86.758839 -46.337940  24.081594   0.052119     0.845984   

                 g_fluxerr      r_mag  r_magerr       r_flux  r_fluxerr  \
coadd_object_id                                                           
931684428         0.238720  16.393660  0.000273  1005.856818   0.252535   
1556041733        0.256970  16.465633  0.000299   941.340283   0.259200   
979002481         0.578791  16.068356  0.000266  1357.243517   0.332330   
995403105         0.180989  16.070789  0.000231  1354.204535   0.288493   
979097465         0.214716  16.111046  0.000255  1304.913319   0.306200   
...                    ...        ...       ...          ...        ...   
887791189         0.143589  25.289423  0.264862     0.278119   0.067846   
1697986932        0.143589  24.895716  0.151750     0.399681   0.055862   
1011682001        0.049464  23.203817  0.040612     1.898773   0.071024   
959369723         0.050111  22.853050  0.027209     2.622889   0.065731   
1580230294        0.040610  22.841181  0.022123     2.651720   0.054031   

                     i_mag  i_magerr      i_flux  i_fluxerr      z_mag  \
coadd_object_id                                                          
931684428        16.711504  0.000417  750.582454   0.288616  15.676861   
1556041733       17.463329  0.000593  375.550437   0.205143  15.783970   
979002481        17.093698  0.000554  527.862244   0.269217  15.902316   
995403105        17.274448  0.000594  446.910588   0.244498  15.892651   
979097465        17.142202  0.000554  504.799653   0.257664  16.109114   
...                    ...       ...         ...        ...        ...   
887791189        22.204578  0.028434    4.766160   0.124820  20.991449   
1697986932       22.037481  0.019640    5.559138   0.100561  20.994463   
1011682001       22.057419  0.024272    5.457987   0.122014  20.994617   
959369723        22.039396  0.023609    5.549342   0.120668  20.998436   
1580230294       22.092491  0.020236    5.284496   0.098494  20.999891   

                 z_magerr       z_flux  z_fluxerr      Y_mag  Y_magerr  \
coadd_object_id                                                          
931684428        0.000259  1946.505683   0.465141  15.332111  0.000570   
1556041733       0.000254  1763.648560   0.412024  15.415310  0.000541   
979002481        0.000335  1581.515905   0.488627  15.743353  0.000668   
995403105        0.000374  1595.657802   0.550371  14.998402  0.000368   
979097465        0.000358  1307.237579   0.431349  15.093253  0.000575   
...                   ...          ...        ...        ...       ...   
887791189        0.017894    14.568682   0.240104  20.781302  0.054016   
1697986932       0.014022    14.528300   0.187630  20.771387  0.038908   
1011682001       0.018400    14.526233   0.246176  20.924448  0.047916   
959369723        0.014383    14.475235   0.191758  20.973963  0.047931   
1580230294       0.014069    14.455845   0.187313  21.399715  0.063653   

                      Y_flux  Y_fluxerr      J_mag  J_magerr       J_flux  \
coadd_object_id                         

In [37]:
neat_cut_df.to_csv('./objs_7102.csv')